dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-etcd/src/main/java/org/apache/dolphinscheduler/plugin/registry/etcd/EtcdRegistry.java

```diff
@@ -0,0 +1,338 @@
+            return keyValues.stream().map(e -> getSubNodeKeyName(prefix, e.getKey().toString(StandardCharsets.UTF_8))).distinct().collect(Collectors.toList());
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RegistryException("etcd get children error", e);
+        } catch (ExecutionException e) {
+            throw new RegistryException("etcd get children error", e);
+        }
+    }
+
+    /**
+     * If "/" exists in the child object, get the string prefixed with "/"
+     */
+    private String getSubNodeKeyName(final String prefix, final String fullPath) {
+        String pathWithoutPrefix = fullPath.substring(prefix.length());
+        return pathWithoutPrefix.contains(FOLDER_SEPARATOR) ? pathWithoutPrefix.substring(0, pathWithoutPrefix.indexOf(FOLDER_SEPARATOR)) : pathWithoutPrefix;
+    }
+
+    @Override
+    public boolean exists(String key) {
+        GetOption getOption = GetOption.newBuilder().withCountOnly(true).build();
+        try {
+            if (client.getKVClient().get(byteSequence(key),getOption).get().getCount() >= 1) {
+                return true;
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RegistryException("etcd check key is existed error", e);
+        } catch (ExecutionException e) {
+            throw new RegistryException("etcd check key is existed error", e);
+        }
+        return false;
+    }
+
+    /**
+     * get the lock with a lease
+     */
+    @Override
+    public boolean acquireLock(String key) {
+        Lock lockClient = client.getLockClient();
+        Lease leaseClient = client.getLeaseClient();
+        // get the lock with a lease
+        try {
+            long leaseId = leaseClient.grant(TIME_TO_LIVE_SECONDS).get().getID();
+            // keep the lease
+            client.getLeaseClient().keepAlive(leaseId, Observers.observer(response -> {
+            }));
+            lockClient.lock(byteSequence(key),leaseId).get();
+
+            // save the leaseId for release Lock
+            if (null == threadLocalLockMap.get()) {
+                threadLocalLockMap.set(new HashMap<>());
+            }
+            threadLocalLockMap.get().put(key,leaseId);
+            return true;
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RegistryException("etcd get lock error", e);
+        } catch (ExecutionException e) {
+            throw new RegistryException("etcd get lock error", e);
+        }
+    }
+
+    /**
+     * release the lock by revoking the leaseId
+     */
+    @Override
+    public boolean releaseLock(String key) {
+        try {
+            Long leaseId = threadLocalLockMap.get().get(key);
+            client.getLeaseClient().revoke(leaseId);
+            threadLocalLockMap.get().remove(key);
```

Comment:
Do we need to check if the `threadLocalLockMap.get()` will null?

Reply:
If this is null, the remove function will throw an exception